tests/providers/tableau/hooks/test_tableau.py

```diff
@@ -189,3 +189,79 @@ def test_get_all(self, mock_pager, mock_server, mock_tableau_auth):
+        """
+        # Test SUCCESS
+        mock_tableau_server.jobs.get_by_id.return_value.finish_code = 0
+        with TableauHook(tableau_conn_id='tableau_test_password') as tableau_hook:
+            tableau_hook.server = mock_tableau_server
+            jobs_status = tableau_hook.get_job_status(job_id='j1')
+            assert jobs_status == TableauJobFinishCode.SUCCESS
+
+        # Test ERROR
+        mock_tableau_server.jobs.get_by_id.return_value.finish_code = 1
+        with TableauHook(tableau_conn_id='tableau_test_password') as tableau_hook:
+            tableau_hook.server = mock_tableau_server
+            jobs_status = tableau_hook.get_job_status(job_id='j1')
+            assert jobs_status == TableauJobFinishCode.ERROR
+
+    @patch('airflow.providers.tableau.hooks.tableau.Server')
+    def test_wait_for_state(self, mock_tableau_server):
+        '''
+        Test wait_for_state
+        '''
+        # Test SUCCESS Positive
+        with TableauHook(tableau_conn_id='tableau_test_password') as tableau_hook:
+            tableau_hook.get_job_status = MagicMock(
+                name='get_job_status',
+                side_effect=[TableauJobFinishCode.PENDING, TableauJobFinishCode.SUCCESS],
+            )
+            assert tableau_hook.wait_for_state(
+                job_id='j1', target_state=TableauJobFinishCode.SUCCESS, check_interval=1
+            )
+
+        # Test SUCCESS Negative
+        with TableauHook(tableau_conn_id='tableau_test_password') as tableau_hook:
+            tableau_hook.get_job_status = MagicMock(
+                name='get_job_status',
+                side_effect=[
+                    TableauJobFinishCode.PENDING,
+                    TableauJobFinishCode.PENDING,
+                    TableauJobFinishCode.ERROR,
+                ],
+            )
+            assert not tableau_hook.wait_for_state(
+                job_id='j1', target_state=TableauJobFinishCode.SUCCESS, check_interval=1
+            )
+
+        # Test ERROR Positive
+        with TableauHook(tableau_conn_id='tableau_test_password') as tableau_hook:
+            tableau_hook.get_job_status = MagicMock(
+                name='get_job_status',
+                side_effect=[
+                    TableauJobFinishCode.PENDING,
+                    TableauJobFinishCode.PENDING,
+                    TableauJobFinishCode.ERROR,
+                ],
+            )
+            assert tableau_hook.wait_for_state(
+                job_id='j1', target_state=TableauJobFinishCode.ERROR, check_interval=1
+            )
+
+        # Test PENDING Positive
+        with TableauHook(tableau_conn_id='tableau_test_password') as tableau_hook:
+            tableau_hook.get_job_status = MagicMock(
+                name='get_job_status',
+                side_effect=[
+                    TableauJobFinishCode.PENDING,
+                    TableauJobFinishCode.PENDING,
+                    TableauJobFinishCode.ERROR,
+                ],
```

Comment:
Why PENDING twice?

Reply:
I wanted to simulate a more realistic case, just to be more secure that everything was ok.